sonaranalyzer-dotnet/its/expected/Automapper/AutoMapper-025B9F93-7953-454B-AF29-E6A55B2AA58F-S3900.json

```diff
@@ -327,19 +327,6 @@
 },
 {
 "id":  "S3900",
-"message":  "Refactor this method to add validation of parameter 'typeMap' before using it.",
-"location":  {
-"uri":  "sources\Automapper\src\AutoMapper\ICtorParamConfigurationExpression.cs",
```

Comment:
I'm not sure this is right... `typeMap` is not really validated above, is it? It's only dereferenced.

But I think this is still fine because we do raise at line 56, so there is a warning for validating `typeMap`

Reply:
Rule S3900 relies on `NullPointerDereference.NullPointerCheck` to check whether a public argument can be null or not. Once a member select on a identifier has been visited, [it will set a non-null constraint](https://github.com/SonarSource/sonar-dotnet/blob/ff7413922f6565eccccda028d0d89f22d2a11c68/sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/NullPointerDereference.cs#L245) on the underlying identifier, in order to avoid duplicate issue.
So this was indeed a FP, as the constraint was no longer compared correctly due to the wrong CFG shape